packages/react-next/src/components/Pivot/useOverflow.ts

```diff
@@ -0,0 +1,127 @@
+  onOverflowItemsChanged: OverflowItemsChangedCallback,
+  rtl: boolean,
+  pinnedIndexes?: readonly number[],
+) => {
+  React.useLayoutEffect(() => {
+    const menuButton = overflowMenuButtonRef?.current;
+    const container = menuButton?.parentElement;
+    const win = container && getWindow(container);
+    if (!container || !menuButton || !win) {
+      return;
+    }
+
+    // items contains the container's children, excluding the overflow menu button itself
+    const items: HTMLElement[] = [];
+    for (let i = 0; i < container.children.length; i++) {
+      const item = container.children[i];
+      if (item instanceof HTMLElement && item !== menuButton) {
+        items.push(item);
+      }
+    }
+
+    // Keep track of the minimum width of the container to fit each child index.
+    // This cache is an integral part of the algorithm and not just a performance optimization: it allows us to
+    // recalculate the overflowIndex on subsequent resizes even if some items are already inside the overflow.
+    const minContainerWidth: number[] = [];
+    let extraWidth = 0; // The accumulated width of items that don't move into the overflow
+
+    const updateOverflow = (containerWidth: number) => {
+      // Iterate the items in reverse order until we find one that fits within the bounds of the container
+      for (let i = items.length - 1; i >= 0; i--) {
+        // Calculate the min container width for this item if we haven't done so yet
+        if (minContainerWidth[i] === undefined) {
+          const itemOffsetEnd = rtl ? containerWidth - items[i].offsetLeft : items[i].offsetLeft + items[i].offsetWidth;
+
+          // If the item after this one is pinned, reserve space for it
+          if (i + 1 < items.length && pinnedIndexes?.includes(i + 1)) {
+            // Use distance between the end of the previous item and this one (rather than the
+            // pinned item's offsetWidth), to account for any margin between the items.
+            extraWidth = minContainerWidth[i + 1] - itemOffsetEnd;
+          }
+
+          // Reserve space for the menu button after the first item was added to the overflow
+          if (i === items.length - 2) {
+            extraWidth += menuButton.offsetWidth;
+          }
+
+          minContainerWidth[i] = itemOffsetEnd + extraWidth;
+        }
+
+        if (containerWidth > minContainerWidth[i]) {
+          setOverflowIndex(i + 1);
+          return;
+        }
+      }
+
+      // If we got here, nothing fits outside the overflow
+      setOverflowIndex(0);
+    };
+
+    let prevOverflowIndex = items.length;
+    const setOverflowIndex = (overflowIndex: number) => {
+      if (prevOverflowIndex !== overflowIndex) {
+        prevOverflowIndex = overflowIndex;
+        onOverflowItemsChanged(
+          overflowIndex,
+          items.map((ele, index) => ({
+            ele,
+            isOverflowing: index >= overflowIndex && !pinnedIndexes?.includes(index),

```

Comment:
use `indexOf` instead of `includes`: https://caniuse.com/#feat=array-includes

Reply:
Thanks for catching that! I'm just going to change it to only support one pinnedIndex to keep it simpler, since the extra complexity of an array of indexes isn't necessary for now.